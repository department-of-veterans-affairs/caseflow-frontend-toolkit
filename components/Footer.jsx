import React from 'react';
import PropTypes from 'prop-types';
import Link from './Link';
import { css } from 'glamor';
import getAppWidthStyling from './util/getAppWidthStyling';

const footerStyles = css({
  marginTop: '100px',
  minHeight: '5em',

  '& > div': {
    padding: '30px 0',
    height: 'auto'
  }
});

export default class Footer extends React.Component {
  onFeedbackClick = () => {
    window.analyticsEvent(this.props.appName, 'feedback', 'footer');
  }

  render() {
    const {
      wideApp,
      feedbackUrl
    } = this.props;

    return <footer className="cf-app-footer" {...footerStyles}>
      <div {...getAppWidthStyling(wideApp)}>
        <div className="cf-push-right">
          <Link
            href={feedbackUrl}
            target="_blank"
            onClick={this.onFeedbackClick}>Send feedback</Link>
        </div>
      </div>
    </footer>;
  }
}

Footer.propTypes = {
  appName: PropTypes.string.isRequired,
  buildDate: PropTypes.string,
  feedbackUrl: PropTypes.string.isRequired
};
