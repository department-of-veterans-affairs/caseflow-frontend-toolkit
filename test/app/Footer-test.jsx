import React from 'react';
// import { expect } from 'chai';
// import { shallow } from 'enzyme';
// import toJson from 'enzyme-to-json';
import { render } from '@testing-library/react';
import Footer from '../../components/Footer';

describe('Footer', () => {
  it('renders with props', () => {
    const { asFragment } = render(
      <Footer feedbackUrl="https://feedback.va.gov" appName="App Name" />
    );

    expect(asFragment()).toMatchSnapshot();
  }
    // expect(
    //   toJson(shallow(<Footer feedbackUrl="https://feedback.va.gov" appName="App Name" />))
    // ).to.matchSnapshot()
  );

  it('respects wideApp setting', () => {
    const { asFragment } = render(
      <Footer feedbackUrl="https://feedback.va.gov" appName="App Name" wideApp />
    );

    expect(asFragment()).toMatchSnapshot();
  }
    // expect(
    //   toJson(shallow(
    //     <Footer feedbackUrl="https://feedback.va.gov" appName="App Name" wideApp />
    //   ))
    // ).to.matchSnapshot()
  );
});
