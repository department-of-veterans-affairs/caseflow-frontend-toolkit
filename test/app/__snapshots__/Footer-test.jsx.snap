// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders with props 1`] = `
<DocumentFragment>
  <footer
    class="cf-app-footer"
    data-css-f8irbh=""
  >
    <div
      data-css-pw7rlt=""
    >
      <div
        class="cf-push-right"
      >
        <a
          href="https://feedback.va.gov"
          target="_blank"
        >
          Send feedback
        </a>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;

exports[`Footer respects wideApp setting 1`] = `
<DocumentFragment>
  <footer
    class="cf-app-footer"
    data-css-f8irbh=""
  >
    <div
      data-css-35ly83=""
    >
      <div
        class="cf-push-right"
      >
        <a
          href="https://feedback.va.gov"
          target="_blank"
        >
          Send feedback
        </a>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
